File: Arrays/majority_element_2.py
#https://www.geeksforgeeks.org/problems/majority-vote/1

class Solution:
    def findMajority(self, arr):
        occurence = {}
        for num in arr:
            if num not in occurence:
                occurence[num] = 1
            else:
                occurence[num] += 1
        
        majority = []
        occurence_sorted = sorted(occurence.items())
        sorted_dict = dict(occurence_sorted)
        for key, value in sorted_dict.items():
            if value > (len(arr)/3):
                majority.append(key)
        return majority

File: Arrays/majority_element_2.java
//https://www.geeksforgeeks.org/problems/majority-vote/1

class Solution {
    public ArrayList<Integer> findMajority(int[] arr) {
        // Code here
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num : arr){
            if(!map.containsKey(num)){
                map.put(num, 1);
            }else{
                map.put(num, map.get(num)+1);
            }
        }
        ArrayList<Integer> majority = new ArrayList<>();
        
        ArrayList<Integer> sortedKeys = new ArrayList<>(map.keySet());
        Collections.sort(sortedKeys);
        
        for(int key: sortedKeys){
            if(map.get(key) > arr.length/3){
                majority.add(key);
            }
        }
        return majority;
    }
}
